Project2015To2017/Transforms/PropertySimplificationTransformation.cs

```diff
@@ -14,189 +14,185 @@ public sealed class PropertySimplificationTransformation : ITransformation
 	{
 		public void Transform(Project definition, ILogger logger)
 		{
-			FilterUnneededProperties(definition, definition.AdditionalPropertyGroups);
-		}
+			if (definition.PrimaryPropertyGroup == null)
+			{
+				throw new InvalidOperationException("Unable to do property simplification pass due to lack of primary PropertyGroup.");
```

Comment:
Not sure if this should be an exception though. Just continue if not found?

Reply:
It was an exception before. And the situation is pretty weird. Even too weird for my liking.